src/middlewared/middlewared/plugins/etc.py

```diff
@@ -266,6 +272,9 @@ def __init__(self, *args, **kwargs):
             if rendered is None:
                 continue
 
+            if not os.path.exists(os.path.dirname(outfile)):
+                os.makedirs(outfile.rsplit('/', 1)[0])
```

Comment:
You replaced only one one them, you're still doing the same thing on makedirs
Should save dirname and reuse it ;)

Reply:
Uh oh, thank you for catching that :)